finetuner/tuner/callback/model_checkpoint.py

```diff
@@ -0,0 +1,141 @@
+import os
+from typing import TYPE_CHECKING, Optional
+import numpy as np
+from jina.logging.logger import JinaLogger
+from finetuner.helper import get_framework
+from .base import BaseCallback
+
+
+if TYPE_CHECKING:
+    from ..base import BaseTuner
+
+
+class ModelCheckpointCallback(BaseCallback):
+    """
+    Callback to save model at every epoch or the best model across all epochs
+    `ModelCheckepointCallback` is used in conjunction with training
+    using `finetuner.fit()`
+
+    A few options this callback provides include:
+    - Whether to only keep the model that has achieved the "best performance" so
+        far, or whether to save the model at the end of every epoch regardless of
+        performance.
+    - Definition of 'best'; which quantity to monitor and whether it should be
+        maximized or minimized.
+    - The frequency it should save at. Currently, the callback supports saving at
+        the end of every epoch, or after a fixed number of training batches.
+    """
+
+    def __init__(
+        self,
+        filepath: str = None,
+        save_best_only: Optional[bool] = False,
+        monitor: Optional[str] = 'val_loss',
+        mode: Optional[str] = 'auto',
+    ):
+        """
+        :param filepath: string or `PathLike`, path to save the model file.
```

Comment:
Can you rename this to something like `save_dir`, because it is the directory where to save. Naming it `filepath` can confuse people into thinking this is the (full) file name of saved model.